openquake/hazardlib/gsim/base.py

```diff
@@ -525,6 +532,37 @@ def to_imt_unit_values(self, values):
         """
         return numpy.exp(values)
 
+    def set_parameters(self):
+        """
+        Combines the parameters of the GMPE provided at the construction level
+        with the ones originally assigned to the backbone modified GMPE.
+        """
+        # Creating the list of keys
+        keys = {}
+        for key in dir(self):
+            if not callable(getattr(self, key)) and not re.search('^_', key):
+                # keys[key] = set(())
+                keys[key] = getattr(self, key)
+        # Setting parameters
+        for key in dir(self.gmpe):
+            if key in keys and not callable(getattr(self.gmpe, key)):
+                if re.search('^[A-Z]', key) and not re.search('^C', key):
+                    tmps = getattr(self.gmpe, key)
+                    try:
+                        keys[key] |= tmps
+                    except TypeError:
+                        if key in ADM_STR or key in ADM_FLOAT:
+                            keys[key] = tmps
+                        elif key in ADM_TAB:
+                            pass
+                        else:
+                            print('This is not a recognized type ', key)
+                            raise
```

Comment:
You can do a `raise NameError(This is not a recognized type ', key)` without `print`.